### PR TITLE
feat: SJIP-725 move newsletter to standalone routes

### PR DESCRIPTION
**Pull request by aperron-ferlab**

### CODE REVIEW

**Review by celinepelletier — APPROVED** [review]

File: src/db/dal/newsletter.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { getNewsletterStatusFetcher, getNewsletterUpdater, SubscriptionStatus } from '../../utils/newsletter';
+import UserModel, { IUserOuput } from '../models/User';
+import { getUserById } from './user';
+
+export const subscribeNewsletter = async (keycloak_id: string, email: string): Promise<IUserOuput> => {
+    const newsletterUpdater = getNewsletterUpdater('includedcc');
```
Comment (celinepelletier) [inline src/db/dal/newsletter.ts:6]: use realm config instead of hardcoded

File: src/db/dal/newsletter.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            newsletter_subscription_status: newsletterStatus,
+            newsletter_email: newsletterStatus !== SubscriptionStatus.FAILED ? email : undefined,
+        });
+    }
+
+    return user;
+};
+
+export const unsubscribeNewsletter = async (keycloak_id: string): Promise<IUserOuput> => {
+    const newsletterUpdater = getNewsletterUpdater('includedcc');
```
Comment (celinepelletier) [inline src/db/dal/newsletter.ts:27]: same

File: src/db/dal/newsletter.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            newsletter_subscription_status: newsletterStatus,
+            newsletter_email: newsletterStatus !== SubscriptionStatus.FAILED ? null : undefined,
+        });
+    }
+
+    return user;
+};
+
+export const refreshNewsletterStatus = async (keycloak_id: string): Promise<IUserOuput> => {
+    const newsletterStatusFetcher = getNewsletterStatusFetcher('includedcc');
```
Comment (celinepelletier) [inline src/db/dal/newsletter.ts:48]: same

File: src/db/dal/user.ts [hunk 4]
```diff
@@ -139,7 +149,7 @@ export const getProfileImageUploadPresignedUrl = async (keycloak_id: string) =>
     };
 };
 
-export const getUserById = async (keycloak_id: string, isOwn: boolean): Promise<IUserOuput> => {
+export const getUserById = async (keycloak_id: string, isOwn: boolean): Promise<UserModel> => {
```
Comment (celinepelletier) [inline src/db/dal/user.ts:152]: a reason to change the return type ?

File: src/utils/newsletter.ts [hunk 5]
```diff
@@ -14,15 +14,28 @@ export type NewsletterPayload = {
     action: SubscriptionStatus;
 };
 
-export interface NewsletterHandler {
-    (payload: NewsletterPayload): Promise<SubscriptionStatus>;
+export interface NewsletterUpdater {
+    (email: NewsletterPayload): Promise<SubscriptionStatus>;
```
Comment (celinepelletier) [inline src/utils/newsletter.ts]: I would name it newsletterPayload instead because it's not just email in it

**Review by celinepelletier — CHANGES_REQUESTED** [review]

File: src/utils/newsletter.ts (hunk 5 above)
Comment (aperron-ferlab) [reply src/utils/newsletter.ts]: Yes I already changed it back to payload (you were on an outdated commit)

File: src/db/dal/user.ts (hunk 4 above)
Comment (aperron-ferlab) [reply src/db/dal/user.ts]: I changed it back to IUserOutput but the real return was UserModel. To keep the IUserOutput, I had to add .dataValues at the end of the return

**Review by atoulous — APPROVED** [review]

**Pull request merged**